extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridStringArrayConverter.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    /// <summary>
+    /// Converter to bind to string[] parameter.
+    /// </summary>
+    [SupportedConverterType(typeof(string[]))]
+    internal class EventGridStringArrayConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.TargetType != typeof(string[]))
+            {
+                return new(ConversionResult.Unhandled());
+            }
+            try
+            {
+                var contextSource = context?.Source as string;
+
+                if (contextSource is null)
+                {
+                    return new(ConversionResult.Failed(new InvalidOperationException("Context source cannot be null")));
```

Comment:
The context source wasn't necessarily null - it may just not have been a string.